pkg/controller/logstorage/logstorage.go

```diff
@@ -161,6 +162,21 @@ func (r *ReconcileLogStorage) createLogStorage(
 		return reconcile.Result{}, false, finalizerCleanup, err
 	}
 
+	// esInternalCertSecret is a secret handled by ECK
+	if esInternalCertSecret != nil {
+		esPublicSecretComponent := render.NewPassthrough(esInternalCertSecret)
+		if install.CertificateManagement != nil {
```

Comment:
The behavior I understand we expect is that for CertificateManagement we need to create the secret because if we allow ECK operator to create it then it is incorrect. So it sounds like ECK operator won't overwrite the secret if it exists so it sounds like we should be owning that Secret.

Since the non-CertificateManagement case also creates this secret if it doesn't exist it sounds like a rare case that ECK will have actually created this cert. So it makes me think we should go ahead and create the Secret with the OwnerReference.
@jaderhs lets just put the OwnerReference on the secret in all cases. That is what happened before so lets keep that behavior for now.
@rene-dekker WDYT?

Reply:
Yes, that works.